bmp180/bmp180.go

```diff
@@ -0,0 +1,191 @@
+// Package bmp180 provides a driver for the BMP180 digital pressure sensor
+// by Bosch
+//
+// Datasheet:
+// https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+//
+package bmp180
+
+import (
+	"time"
+
+	"github.com/aykevl/tinygo/src/machine"
+)
+
+const (
+	// ULTRALOWPOWER is the lowest oversampling mode of the pressure measurement.
+	ULTRALOWPOWER OversamplingMode = iota
+	// BSTANDARD is the standard oversampling mode of the pressure measurement.
+	STANDARD
+	// HIGHRESOLUTION is a high oversampling mode of the pressure measurement.
+	HIGHRESOLUTION
+	// ULTRAHIGHRESOLUTION is the highest oversampling mode of the pressure measurement.
+	ULTRAHIGHRESOLUTION
+)
+
+// BMP180OversamplingMode is the oversampling ratio of the pressure measurement.
+type OversamplingMode uint
+
+// calibrationCoefficients reads at startup and stores the calibration coefficients
+type calibrationCoefficients struct {
+	ac1 int16
+	ac2 int16
+	ac3 int16
+	ac4 uint16
+	ac5 uint16
+	ac6 uint16
+	b1  int16
+	b2  int16
+	mb  int16
+	mc  int16
+	md  int16
+}
+
+// Device wraps an I2C connection to a BMP180 device.
+type Device struct {
+	bus                     machine.I2C
+	mode                    OversamplingMode
+	calibrationCoefficients calibrationCoefficients
+}
+
+// New creates a new BMP180 connection. The I2C bus must already be
+// configured.
+//
+// This function only creates the Device object, it does not touch the device.
+func New(bus machine.I2C) Device {
```

Comment:
AFAIK it's good practise to include type name in "constructor"

Reply:
Not always, consider `errors.New()`. The main thing about naming things in Go is not repeating anything, and in this case every package is one device. You don't have to repeat what it is. But of course, that's subjective :)
In any case, I care most about consistency and all other drivers use a bare `New` function.

Reply:
Yes, do not repeat especially the package name. That's subjective of course and it's what I can live with 👍